similar/main/mission.cpp

```diff
@@ -1088,13 +1088,87 @@ int select_mission(int anarchy_mode, const char *message, window_event_result (*
 }
 
 #if DXX_USE_EDITOR
+int write_mission(void)
+{
+	auto &msn_extension =
+#if defined(DXX_BUILD_DESCENT_II)
+	(Current_mission->descent_version == Mission::descent_version_type::descent2) ? MISSION_EXTENSION_DESCENT_II :
+#endif
+	MISSION_EXTENSION_DESCENT_I;
+	array<char, PATH_MAX> mission_filename;
+	snprintf(mission_filename.data(), mission_filename.size(), "%s%s", Current_mission->path.c_str(), msn_extension);
+	
+	auto &&mfile = PHYSFSX_openWriteBuffered(mission_filename.data());
+	if (!mfile)
+	{
+		PHYSFS_mkdir(MISSION_DIR);	//try making directory - in *write* path
+		mfile = PHYSFSX_openWriteBuffered(mission_filename.data());
+		if (!mfile)
+			return 0;
+	}
+
+#if defined(DXX_BUILD_DESCENT_II)
+	switch (Current_mission->descent_version)
+	{
+		case Mission::descent_version_type::descent2x:
+			PHYSFSX_printf(mfile, "x");
+			break;
+
+		case Mission::descent_version_type::descent2z:
+			PHYSFSX_printf(mfile, "z");
+			break;
+			
+		case Mission::descent_version_type::descent2a:
+			PHYSFSX_printf(mfile, "!");
+			break;
+
+		default:
+			break;
+	}
+#endif
+
+	PHYSFSX_printf(mfile, "name = %s\n", static_cast<const char *>(Current_mission->mission_name));
+
+	PHYSFSX_printf(mfile, "type = %s\n", Current_mission->anarchy_only_flag ? "anarchy" : "normal");
+
+	if (Briefing_text_filename[0])
+		PHYSFSX_printf(mfile, "briefing = %s\n", static_cast<const char *>(Briefing_text_filename));
+
+	if (Ending_text_filename[0])
+		PHYSFSX_printf(mfile, "ending = %s\n", static_cast<const char *>(Ending_text_filename));
+
+	PHYSFSX_printf(mfile, "num_levels = %i\n", Last_level);
+
+	range_for (auto &i, unchecked_partial_range(Level_names.get(), Last_level))
+		PHYSFSX_printf(mfile, "%s\n", static_cast<const char *>(i));
+
+	if (N_secret_levels)
+	{
+		PHYSFSX_printf(mfile, "num_secrets = %i\n", N_secret_levels);
+
+		for (int i = 0; i < N_secret_levels; i++)
+			PHYSFSX_printf(mfile, "%s,%i\n", static_cast<const char *>(Secret_level_names[i]), Secret_level_table[i]);
+	}
+
+#if defined(DXX_BUILD_DESCENT_II)
+	if (Current_mission->alternate_ham_file)
+		PHYSFSX_printf(mfile, "ham = %s\n", static_cast<const char *>(*Current_mission->alternate_ham_file.get()));
+#endif
+
+	mfile.reset();
```

Comment:
I think in `load_mission` it is explicitly reset because the non-automatic version closed `mfile` before running some other statements.  In that case, I added the explicit reset to maintain the ordering, although it is probably not necessary.  In your commit, it is effectively the last statement, so the reset happens in the same place whether or not it is explicit.